apps/bridge/src/utils/hooks/useCCTPBridgeStatus.ts

```diff
@@ -0,0 +1,131 @@
+  const [messageHash, setMessageHash] = useState<`0x${string}` | undefined>(undefined);
+  const publicClient = usePublicClient({
+    chainId: bridgeDirection === 'deposit' ? l2ChainID : l1ChainID,
+  });
+
+  const { data: initiateTxReceipt } = useWaitForTransaction({
+    hash: initiateTxHash,
+    chainId: bridgeDirection === 'deposit' ? l1ChainID : l2ChainID,
+  });
+
+  const { data: bridgeAttestation } = useQuery(
+    ['bridgeAttestation', { initiateTxHash, messageHash }],
+    async () => fetchBridgeAttestation(messageHash as string),
+    {
+      enabled: !!messageHash,
+      refetchOnMount: false,
+      refetchOnWindowFocus: false,
+      placeholderData: undefined,
+    },
+  );
+
+  useEffect(() => {
+    if (initiateTxReceipt) {
+      if (initiateTxReceipt.status === 'reverted') {
+        setStatus('INITIATE_CCTP_BRIDGE_FAILED');
+        return;
+      }
+
+      const eventTopic = getEventSelector('MessageSent(bytes)');
+      const log = initiateTxReceipt.logs.find((l) => l.topics[0] === eventTopic);
+      const messageBytes = decodeAbiParameters(
+        [parseAbiParameter('bytes message')],
+        log?.data as `0x${string}`,
+      )[0];
+      const hash = keccak256(messageBytes);
+      setMessage(messageBytes);
+      setMessageHash(hash);
+    }
+  }, [initiateTxReceipt]);
+
+  useEffect(() => {
+    if (bridgeAttestation?.attestation && message) {
+      void (async () => {
+        if (bridgeAttestation.status === 'pending_confirmations') {
+          setStatus('INITIATE_CCTP_BRIDGE_PENDING');
+          return;
+        }
+
+        // We still need to check whether or not the message was received on the other chain.
+        // If it hasn't been received yet, the user needs to call receiveMessage on the MessageTransmitter.
+        // It it has been received, the bridge is complete.
+        // We can check if the message has been received by simulating a call to receiveMessage.
+        // If it fails, assume it's because the message has already been received.
+        try {
+          await publicClient.simulateContract({
+            address:
+              bridgeDirection === 'deposit'
+                ? publicRuntimeConfig.l2CCTPMessageTransmitterAddress
+                : publicRuntimeConfig.l1CCTPMessageTransmitterAddress,
+            abi: MessageTransmitter,
+            functionName: 'receiveMessage',
+            args: [message, bridgeAttestation.attestation],
+          });
+          // Success, ie the message still needs to be received.
+          setStatus('FINALIZE_CCTP_BRIDGE');
+        } catch (e) {
+          // Failed, ie the message has already been received.
+          setStatus('CCTP_BRIDGE_COMPLETE');
+        }
+      })();
+    }
+  }, [bridgeAttestation, bridgeDirection, initiateTxReceipt, message, publicClient]);
+
+  // We need to be able to set the status from the FinzliaeCCTPBridgeButton, so we expose a setter.
+  const handleSetStatus = useCallback(
```

Comment:
This doesn't seem to do anything different than `setStatus` so I believe U can just directly return `setStatus` below.